packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -15,48 +15,48 @@
  */
 
 import { randomTraceId } from '@opencensus/web-core';
+import { AsyncTask } from './zone-types';
+import {
+  OnPageInteractionStopwatch,
+  startOnPageInteraction,
+} from './on-page-interaction';
 
 // Allows us to monkey patch Zone prototype without TS compiler errors.
 declare const Zone: ZoneType & { prototype: Zone };
 
-export interface AsyncTaskData {
-  interactionId: string;
-  pageView: string;
-}
-
-export type AsyncTask = Task & {
-  data: AsyncTaskData;
-  eventName: string;
-  target: HTMLElement;
-  // Allows access to the private `_zone` property of a Zone.js Task.
-  _zone: Zone;
-};
-
 // Delay of 50 ms to reset currentEventTracingZone.
 const RESET_TRACING_ZONE_DELAY = 50;
 
 export class InteractionTracker {
   // Allows to track several events triggered by the same user interaction in the right Zone.
   private currentEventTracingZone?: Zone;
 
+  // Map of interaction Ids to stopwatches.
+  private readonly interactions: {
+    [index: string]: OnPageInteractionStopwatch;
+  } = {};
+
+  // Map of interaction Ids to timeout ids.
+  private readonly interactionCompletionTimeouts: {
+    [index: string]: number;
+  } = {};
+
   constructor() {
     const runTask = Zone.prototype.runTask;
     Zone.prototype.runTask = (
       task: AsyncTask,
       applyThis: unknown,
       applyArgs: unknown
     ) => {
-      const time = Date.now();
-
       console.warn('Running task');
       console.log(task);
       console.log(task.zone);
 
+      const interceptingElement = isTrackedElement(task);
```

Comment:
Since the function is returning the element now, what would you think about calling it `getTrackedElement`?